homeassistant/components/esphome/button.py

```diff
@@ -32,6 +31,11 @@ async def async_setup_entry(
 class EsphomeButton(EsphomeEntity[ButtonInfo, EntityState], ButtonEntity):
     """A button implementation for ESPHome."""
 
+    @property
+    def device_class(self) -> str:
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return self._static_info.device_class
```

Comment:
Is this validated somewhere that it's one of our standard device classes. That's a requirement if we don't use the constants directly.

Reply:
> Is this validated somewhere that it's one of our standard device classes. That's a requirement if we don't use the constants directly.

Yes, it is limited/validated @ ESPHome.

https://github.com/esphome/esphome/pull/2835/files#diff-39f2d661abcb3c0293c7736f20ab42715d1c456e0c74c48b69574f4ae6eecfa8R44

(As done with other device classes, state classes or entity categories)

Reply:
It must be validated inside Home Assistant since it's the API of Home Assistant. Eg:

https://github.com/home-assistant/core/blob/64afe738cccabd8473ec8a54adddeb6c186772e9/homeassistant/components/esphome/sensor.py#L99-L104

Reply:
The state class is also using a map in Home Assistant:
https://github.com/home-assistant/core/blob/64afe738cccabd8473ec8a54adddeb6c186772e9/homeassistant/components/esphome/sensor.py#L119

Reply:
👍  Alright, let's do the same here in that case

Reply:
#60594